jujugui/static/gui/src/app/app.js

```diff
@@ -1892,9 +1892,14 @@ YUI.add('juju-gui', function(Y) {
       @param {String} uuid The environment UUID where to switch to.
       @param {String} username The username for the new environment.
       @param {String} password The password for the new environment.
+      @param {Boolean} reconnect Whether to reconnect to a new environment.
     */
-    switchEnv: function(socketUrl, username, password) {
-      console.log('switching to new socket URL:', socketUrl);
+    switchEnv: function(socketUrl, username, password, reconnect=!!socketUrl) {
```

Comment:
https://github.com/juju/juju-gui/pull/1509#discussion-diff-58904948

I'll let you and @hatched battle this one out :-)

Just to be clear: we reconnect only if there *is* a socketUrl.